packages/discussions/src/utils/posts/index.ts

```diff
@@ -73,7 +73,7 @@ export function canDeletePost(
 }
 
 function isPostCreator(post: IPost, user: IUser) {
-  return post.creator === user.username;
+  return !!user.username && post.creator === user.username;
```

Comment:
Maybe outside the scope of this PR, but I wonder if these discussions utils that receive an `IUser` object shouldn't be updated to accept `undefined|null` since we won't have a user record in the anonymous use case. I could see their `user` signatures being updated to `user?: IUser | null`.

Reply:
👍